neuron-minestom/src/main/java/com/github/phantazmnetwork/neuron/bindings/minestom/entity/goal/GoalGroup.java

```diff
@@ -20,10 +36,32 @@ public class GoalGroup implements Tickable {
      *
      * @param goals The {@link NeuralGoal}s in the group
      */
-    public GoalGroup(@NotNull Iterable<NeuralGoal> goals) {
+    @FactoryMethod
+    public GoalGroup(@NotNull Data data, @NotNull @DataName("goals") Collection<NeuralGoal> goals) {
+        this(goals);
+    }
+
+    public GoalGroup(@NotNull Collection<NeuralGoal> goals) {
         this.goals = Objects.requireNonNull(goals, "goals");
     }
 
+    @ProcessorMethod
+    public static @NotNull ConfigProcessor<Data> processor() {
+        ConfigProcessor<Collection<String>> pathProcessor = ConfigProcessor.STRING.collectionProcessor();
```

Comment:
should probably be private static + moved inside the anonymous class

Reply:
why private static? and what anonymous class?

Reply:
the one right below this line